src/components/EditOrderForm.js

```diff
@@ -78,69 +83,70 @@ export function EditOrderForm({
-            </RadioGroup>
-          </FormControl>
-
+            {deliveryMethod === 'delivery' && user.userType === 'admin' && (
+              <FormControlLabel
+                control={<Switch checked={haveExtraFee} onChange={() => setHaveExtraFee(!haveExtraFee)} inputProps={{ 'aria-label': 'controlled' }} />}
+                label='Definir taxa de entrega'
+                sx={{ width: '100%', pt: { xs: 0, md: 3 } }}
+              />
+            )}
+          </Box>
           {!isAddressNotVisible && (
-            <TextField
-              label='Morada'
-              type='text'
-              variant='outlined'
-              fullWidth
-              required={requiredInput}
-              sx={editOrderClasses.formField}
-              onChange={(e) => setFullAddress(e.target.value)}
-              placeholder='Rua dos bolos n 7'
-              error={contactError}
-              value={fullAddress}
-              ref={addressRef}
-            />
-          )}
-
-          {user._id !== order.userId._id && (
-            <TextField
-              id='outlined-read-only-input'
-              label='Mensagem'
-              defaultValue={message}
-              InputProps={{
-                readOnly: true,
-              }}
-              sx={editOrderClasses.formTextArea}
-              multiline
-              maxRows={4}
-            />
+            <>
+              {user.userType === 'admin' && haveExtraFee && (
+                <TextField
+                  label='Taxa de entrega'
+                  type='text'
+                  variant='outlined'
+                  fullWidth
+                  required
+                  sx={editOrderClasses.formField}
+                  onChange={(e) => validateDeliveryFee(e)}
+                  error={deliveryPriceError}
+                  value={deliveryPrice}
+                />
+              )}
+              <TextField
+                label='Morada'
+                type='text'
+                variant='outlined'
+                fullWidth
+                required={requiredInput}
+                sx={editOrderClasses.formField}
+                onChange={(e) => setFullAddress(e.target.value)}
+                placeholder='Rua dos bolos n 7'
+                error={addressError}
+                value={fullAddress}
+                ref={addressRef}
+              />
+            </>
           )}
 
-          {user._id === order.userId._id && (
-            <TextField
-              id='outlined-multiline-flexible'
-              label='Mensagem'
-              multiline
-              maxRows={4}
-              sx={editOrderClasses.formTextArea}
-              onChange={(e) => setMessage(e.target.value)}
-              value={message}
-              placeholder='Escreva aqui a sua mensagem...'
-            />
-          )}
+          <TextField
+            id='outlined-multiline-flexible'
```

Comment:
do you need id? asking out of curiosity

Reply:
I don't think so. I believe it got there because i probably copied the code from somewhere. oops!